src/js/tracks/vtt.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * @file vtt.js
+ */
+import vttjs from 'videojs-vtt.js';
+import Component from '../component.js';
+import mergeOptions from '../utils/merge-options.js';
+import window from 'global/window';
+import document from 'global/document';
+import {isPlainEmpty} from '../utils/obj.js';
+
+let loadedCallback;
+
+export let vttjsLoaded = !isPlainEmpty(vttjs) ||
+                         typeof window.WebVTT === 'function';
+
+/**
+ * The `VttLoader` is the `Component` that decides how to load in vttjs
+ *
+ * @extends Component
+ * @fires VttLoader#vttjsloaded
+ * @fires VttLoader#vttjserror
+ */
+class VttLoader extends Component {
+  constructor(player, options, ready) {
+    const options_ = mergeOptions({createEl: false}, options);
+
+    super(player, options_, ready);
+
+    if (vttjsLoaded) {
+      this.setVttjs(vttjs);
+      this.triggerLoaded();
+    } else {
+      this.setVttjs(null);
+
+      player.on('ready', () => {
+        if (!player.tech_.featuresNativeTextTracks) {
+          this.loadRemoteVtt();
+        }
+      });
+    }
+  }
+
+  setVttjs(newVttjs) {
+    VttLoader.vttjs = newVttjs;
+  }
+
+  triggerLoaded() {
+    vttjsLoaded = true;
+
+    /**
+     * Fired when vtt.js is loaded.
+     *
+     * @event VttLoader#vttjsloaded
+     * @type {EventTarget~Event}
+     */
+    this.trigger('vttjsloaded');
+
+    this.player_.ready(() => {
+      /**
+       * Fired when vtt.js is loaded.
+       *
+       * @event Tech#vttjsloaded
+       * @type {EventTarget~Event}
+       */
+      this.player_.tech_.trigger('vttjsloaded');
+    }, true);
+
+    if (loadedCallback) {
+      // eslint-disable-next-line no-use-before-define
+      loadedCallback(getVttjs());
+    }
+  }
+
+  triggerError() {
+    vttjsLoaded = false;
+
+    /**
+     * Fired when vtt.js was not loaded due to an error
+     *
+     * @event VttLoader#vttjserror
+     * @type {EventTarget~Event}
+     */
+    this.trigger('vttjserror');
```

Comment:
Do we want to trigger this event here? seems linke we only listen for this on tech